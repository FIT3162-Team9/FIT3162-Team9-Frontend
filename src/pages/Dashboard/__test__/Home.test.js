import React from 'react';
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Home from './..'
import { render } from '@testing-library/react';
import CircularProgressWithLabel,{bushfireColor} from './../Circle';

it("Home renders without crashing", () => {
  shallow(<Home LGA='banyule' station='86036' />);
});

it("Home renders correctly", () => {
  const tree = shallow(<Home LGA='banyule' station='86036' />);
  expect(toJson(tree)).toMatchSnapshot();
});


describe("Location displayed correctly", () => {
  it("LGA and Station ID displayed correctly on screen", () => {
    const {getByText}= render(<Home LGA='Banyule' station='76031' />);
    expect(getByText(/Banyule/i)).toBeInTheDocument();
  });
});

describe('Circle display correct data', () => {

  it('Date displayed correctly', () =>{
    const {getByText} = render(<CircularProgressWithLabel day={17} month={3} bushfirerating={19.97}/>);
    expect(getByText('17/3')).toBeInTheDocument();  
  });

  it('Bushfire rating displayed correctly', () =>{
    const {getByText} = render(<CircularProgressWithLabel day={17} month={3} bushfirerating={19.97}/>);
    expect(getByText('19.9')).toBeInTheDocument();  
  });

  it('Color returned correctly', () =>{
    expect(bushfireColor(80)).toBe('#ff4040');
  });
});