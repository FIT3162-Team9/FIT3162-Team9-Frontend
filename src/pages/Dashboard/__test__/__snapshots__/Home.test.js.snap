// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalysisChart renders correctly 1`] = `
<WithStyles(ForwardRef(Container))
  className="makeStyles-container-3"
  maxWidth="lg"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Paper))
      className="makeStyles-outerPaper-8"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        lg={9}
        md={8}
        xs={12}
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-headerFont-9"
        >
          Selected LGA: 
          Banyule
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-headerFont-9"
        >
          Selected Station: 
          86036
        </WithStyles(ForwardRef(Typography))>
        <DateRangePicker
          calendarIcon={
            <svg
              className="react-daterange-picker__calendar-button__icon react-daterange-picker__button__icon"
              height={19}
              stroke="black"
              strokeWidth={2}
              viewBox="0 0 19 19"
              width={19}
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="15"
                width="15"
                x="2"
                y="2"
              />
              <line
                x1="6"
                x2="6"
                y1="0"
                y2="4"
              />
              <line
                x1="13"
                x2="13"
                y1="0"
                y2="4"
              />
            </svg>
          }
          className="makeStyles-dateFont-10"
          clearIcon={
            <svg
              className="react-daterange-picker__clear-button__icon react-daterange-picker__button__icon"
              height={19}
              stroke="black"
              strokeWidth={2}
              viewBox="0 0 19 19"
              width={19}
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="4"
                x2="15"
                y1="4"
                y2="15"
              />
              <line
                x1="15"
                x2="4"
                y1="4"
                y2="15"
              />
            </svg>
          }
          closeCalendar={true}
          isOpen={null}
          name="daterange"
          onChange={[Function]}
          rangeDivider="â€“"
          value={
            Array [
              "2019-09-23T01:58:42.969Z",
              "2020-09-23T01:58:42.969Z",
            ]
          }
        />
        <WithStyles(ForwardRef(Button))
          className="makeStyles-addButton-11"
          onClick={[Function]}
        >
          Add User Log
        </WithStyles(ForwardRef(Button))>
        <WithStyles(ForwardRef(Button))
          onClick={[Function]}
        >
          Clear
        </WithStyles(ForwardRef(Button))>
        <WithStyles(ForwardRef(Divider)) />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Paper))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Container))>
`;
